server/controllers/mealsController.js

```diff
@@ -0,0 +1,87 @@
+const connection = require("../dbConnection");
+const mealSchema = require("../models/meal");
+
+const Meal = connection.model("Meal", mealSchema);
+
+const findOneWithId = async (id) => {
+    const { _doc } = await Meal.findById(id);
+    return _doc;
+};
+
+// create
+const create = async ({
+    title,
+    picURL,
+    price,
+    servingSize,
+    servingType,
+    cuisineType,
+    ingredients,
+    requirements,
+    createdAt,
+    updatedAt,
+    chefId,
+}) => {
+    const { _doc } =
+        (await Meal.create({
+            title,
+            picURL,
+            price,
+            servingSize,
+            servingType,
+            cuisineType,
+            ingredients,
+            requirements,
+            createdAt,
+            updatedAt,
+            chefId,
+        })) || {};
+    return _doc;
+};
+
+// update
+const update = async ({
+    id,
+    title,
+    picURL,
+    price,
+    servingSize,
+    servingType,
+    cuisineType,
+    ingredients,
+    requirements,
+    createdAt,
+    updatedAt,
+    chefId,
+}) => {
+    const meal = await Meal.findById(id);
```

Comment:
Do we need a try catch in case the id is invalid?

Reply:
Much earlier in our team stand-ups we decided to do validations in route itself.
So make it work for invalid meal `id` I'm using the following code in the route for `PUT /meals/:id`

```js
        const meal = await mealController.findOneWithId(id);
        if (!meal) {
            res.status(400).json({ errors: ["Meal for id not found"] });
            return;
        }

        const newMeal = await mealController.update({
            id,
            title,
            picURL,
            price,
            servingSize,
            servingType,
            cuisineType,
            ingredients,
            requirements,
            chefId,
        });
```

It should work for the most part, unless the meal get deleted right after the check in route but before the update is called.

I can move more of our validation closer to the models itself, which'll make things DRYer and cover all edge cases.

Thoughts & comments: @sagar-13 @megangilpin @sina-jamshidi

Reply:
Oh I missed that. This should be fine as is!